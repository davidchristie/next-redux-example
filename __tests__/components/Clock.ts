import Clock from "../../components/Clock";
import {
  describeComponent,
  itShouldMatchSnapshot,
  itShowsText
} from "../../test-utils";

describe("At 18:05:31", () => {
  const lastUpdate = new Date(
    "Fri Jan 31 2020 19:05:31 GMT+0100 (Central European Standard Time)"
  ).getTime();

  describeComponent(Clock, { lastUpdate, light: false }, getContext => {
    itShowsText(getContext, "18:05:31");

    itShouldMatchSnapshot(getContext);
  });

  describe("Light", () => {
    describeComponent(Clock, { lastUpdate, light: true }, getContext => {
      itShowsText(getContext, "18:05:31");

      itShouldMatchSnapshot(getContext);
    });
  });
});
