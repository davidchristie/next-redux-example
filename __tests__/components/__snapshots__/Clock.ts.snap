// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`At 18:05:31 Clock Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-1453841751 "
  >
    18:05:31
  </div>
</DocumentFragment>
`;

exports[`At 18:05:31 Light Clock Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-1453841751 light"
  >
    18:05:31
  </div>
</DocumentFragment>
`;
